zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ExporterBatchWriter.java

```diff
@@ -58,6 +73,7 @@ private void updateAndCacheEntity(
 
     final var entity = entityAndHandlers.entity;
     handler.updateEntity(record, entity);
+    cachedRecordTimestamps.put(record.getPosition(), record.getTimestamp());
```

Comment:
:thought_balloon: This has changed from before, we now only track handled records, and using the position as "cache" in case we retry.